File_Adapter/FileAdapter.cs

```diff
@@ -0,0 +1,134 @@
+            "\nWhen Pushing, this is used for pushing objects that are not BHoM `File` or `Directory`." +
+            "\nWhen Pulling, if no request is specified, a FileContentRequest is automatically generated with this location." +
+            "\nBy default this is `C:\\temp\\Filing_Adapter-objects.json`.")]
+        [PreviousVersion("4.0", "BH.Adapter.Filing.FilingAdapter(System.String)")]
+        public FileAdapter(string targetLocation)
+        {
+            Init(targetLocation);
+        }
+
+        [PreviousVersion("4.0", "BH.Adapter.FileAdapter.FileAdapter(System.String, System.String)")]
+        public FileAdapter(string folder, string fileName)
+        {
+            if (folder?.Count() > 2 && folder?.ElementAt(1) != ':')
+                folder = Path.Combine(@"C:\ProgramData\BHoM\DataSets", folder);
+
+            string location = Path.Combine(folder, fileName);
+
+            Init(location);
+        }
+
+        /***************************************************/
+        /**** Private Fields                            ****/
+        /***************************************************/
+
+        private bool m_Push_enableDeleteWarning = true;
+        private bool m_Remove_enableDeleteWarning = true;
+        private bool m_Remove_enableDeleteAlert = true;
+        private bool m_Execute_enableWarning = true;
+        private string m_defaultFilePath = null;
+
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        // Initialisation method for when the File Adapter is instantiated with a location.
+        private bool Init(string location)
+        {
+            if (string.IsNullOrWhiteSpace(location))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Please specifiy a valid target location.");
+                return true;
+            }
+
+            m_defaultFilePath = location;
+
+            if (!ProcessExtension(ref m_defaultFilePath))
+                return false;
+
+            BH.Engine.Reflection.Compute.RecordNote($"The adapter will always target the input location `{location}`." +
+                $"\nTo target multiple Files, use the {this.GetType().Name} constructor with no input.");
+
+            // By default, the objects are appendend to the file if it exists already.
+            this.m_AdapterSettings.DefaultPushType = oM.Adapter.PushType.CreateOnly;
+
+            return true;
+        }
+
+        /***************************************************/
+
+        // Checks on the file extension.
+        private bool ProcessExtension(ref string filePath)
+        {
+            string ext = Path.GetExtension(filePath);
+
+            if (!Path.HasExtension(m_defaultFilePath))
+            {
+                Engine.Reflection.Compute.RecordWarning($"No extension specified in the FileName input. Defaulting to .json.");
```

Comment:
```suggestion
                Engine.Reflection.Compute.RecordNote($"No extension specified in the FileName input. Defaulting to .json.");
```

Reply:
@alelom I would downgrade this here as think this might be a prevalent assumption in File Adapter usage across scripts - and think fine to not raise as warning

Reply:
Sorry - forgot to change that. Applied suggestion.